src/Other/Helpers/JSON.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+namespace Rubix\ML\Other\Helpers;
+
+use RuntimeException;
+
+use const JSON_ERROR_NONE;
+use const JSON_ERROR_SYNTAX;
+use const JSON_ERROR_DEPTH;
+use const JSON_ERROR_STATE_MISMATCH;
+use const JSON_ERROR_CTRL_CHAR;
+use const JSON_ERROR_UTF8;
+
+/**
+ * JSON
+ *
+ * A helper class providing functions relating to json.
+ *
+ * @category    Machine Learning
+ * @package     Rubix/ML
+ * @author      Chris Simpson
+ */
+class JSON
+{
+    protected const DEFAULT_DEPTH = 512;
+
+    protected const DEFAULT_OPTIONS = 0;
+
+    /**
+     * Deserialize JSON.
+     *
+     * @param string $json
+     * @param int $options
+     * @param int $depth
+     *
+     * @throws \RuntimeException
+     *
+     * @return array<mixed>
+     */
+    public static function decode(string $json, int $options = self::DEFAULT_OPTIONS, int $depth = self::DEFAULT_DEPTH) : array
+    {
+        $value = json_decode($json, true, $depth, $options);
+        $error = json_last_error();
+
+        switch ($error) {
+            case JSON_ERROR_NONE:
+                return $value;
+            case JSON_ERROR_SYNTAX:
+                throw new RuntimeException('Malformed JSON Provided. Syntax Error.');
+            case JSON_ERROR_DEPTH:
+                throw new RuntimeException('Maximum stack depth exceeded.');
+            case JSON_ERROR_STATE_MISMATCH:
+                throw new RuntimeException('Underflow or the modes mismatch.');
+            case JSON_ERROR_CTRL_CHAR:
+                throw new RuntimeException('Unexpected control character found.');
+            case JSON_ERROR_UTF8:
```

Comment:
Some of the possible errors are not listed here such as NAN and INF. Should we catch those errors or did you intend for the default case to handle those?

Reply:
👍 Yeah, let me add those in now